src/dex/fluid-dex/fluid-dex-pool.ts

```diff
@@ -26,19 +33,138 @@ export class FluidDexEventPool extends StatefulEventSubscriber<FluidDexPoolState
     protected network: number,
     protected dexHelper: IDexHelper,
     logger: Logger,
-    protected fluidDexIface = new Interface(FluidDexABI), // TODO: add any additional params required for event subscriber
+    protected liquidityIface = new Interface(LiquidityABI), // TODO: add any additional params required for event subscriber
   ) {
     // TODO: Add pool name
     super(parentName, pool.id, dexHelper, logger);
 
     // TODO: make logDecoder decode logs that
-    this.logDecoder = (log: Log) => this.fluidDexIface.parseLog(log);
-    this.addressesSubscribed = [pool.address, pool.token0, pool.token1];
+    this.logDecoder = (log: Log) => this.liquidityIface.parseLog(log);
+    this.addressesSubscribed = [pool.liquidityUserModule];
 
     // Add handlers
-    // this.handlers['Swap'] = this.handleSwap.bind(this);
+    this.handlers['LogOperate'] = this.handleOperate.bind(this);
   }
 
+  /**
+   * Handle a trade rate change on the pool.
+   */
+  async handleOperate(
+    event: any,
+    state: DeepReadonly<FluidDexPoolState>,
+    log: Readonly<Log>,
+  ): Promise<DeepReadonly<FluidDexPoolState> | null> {
+    const ResolverAbi = new Interface(ResolverABI);
+    if (
+      !(
+        event.args.user in
+        [
+          this.pool.address,
+          this.pool.colOperations,
+          this.pool.debtOperations,
+          this.pool.perfectOperationsAndSwapOut,
+        ]
+      )
+    ) {
+      return null;
+    }
+    const callData: MultiCallParams<PoolWithReserves>[] = [
+      {
+        target: this.pool.resolver,
+        callData: ResolverAbi.encodeFunctionData('getPoolReserves', [
+          this.pool.address,
+        ]),
+        decodeFunction: await this.decodePoolWithReserves,
+      },
+    ];
+
+    const results: PoolWithReserves[] =
+      await this.dexHelper.multiWrapper.aggregate<PoolWithReserves>(
+        callData,
+        await this.dexHelper.provider.getBlockNumber(),
+        this.dexHelper.multiWrapper.defaultBatchSize,
+      );
+
+    return {
+      collateralReserves: results[0].collateralReserves,
+      debtReserves: results[0].debtReserves,
+    };
+  }
+
+  decodePoolWithReserves = (
+    result: MultiResult<BytesLike> | BytesLike,
+  ): PoolWithReserves => {
+    const [isSuccess, toDecode] = this.extractSuccessAndValue(result);
+
+    // if (!isSuccess) {
+    //   return null;
+    // }
+
+    const decodedResult = defaultAbiCoder.decode(
```

Comment:
is this how other projects decode too? can't we use the ABI that we have added as json file?

Reply:
I have now used the `generalDecoder` which is used by many other protocols